### PR TITLE
Improve load performance

### PR DESCRIPTION
**Pull request by yah01**
Load sealed segment's fields concurrently

Signed-off-by: yah01 <yang.cen@zilliz.com>
#15736

### CODE REVIEW

File: internal/querynode/segment_loader.go [hunk 1]
```diff
@@ -328,14 +335,64 @@ func (loader *segmentLoader) loadFiledBinlogData(segment *Segment, fieldBinlogs
...
+}
+
+// async load field of sealed segment
+func (loader *segmentLoader) loadSealedFieldAsync(segment *Segment, field *datapb.FieldBinlog) *concurrency.Future {
+	iCodec := storage.InsertCodec{}
+
+	// Avoid consuming too much memory if no CPU worker ready,
+	// acquire a CPU worker before load field binlogs
+	return loader.cpuPool.Submit(func() (interface{}, error) {
+		futures := loader.loadFieldBinlogsAsync(field)
```
Comment (xiaofan-luan) [inline internal/querynode/segment_loader.go:374]: should move the IO part out of cpuPool?

Comment (xiaofan-luan) [inline internal/querynode/segment_loader.go:374]: should move the IO part out of cpuPool?

**Review by xiaofan-luan — COMMENTED** [review]
overall lgtm~

File: internal/querynode/segment_loader.go (hunk 1 above)
Comment (yah01) [reply internal/querynode/segment_loader.go:374]: That may lead to accumulate many binlogs in memory

**Review by letian-jiang — APPROVED** [review]
/lgtm

**Pull request merged**